src/Sentry/Internal/MainExceptionProcessor.cs

```diff
@@ -49,7 +56,30 @@ public void Process(Exception exception, SentryEvent sentryEvent)
 
                 foreach (var key in sentryException.Data.Keys)
                 {
-                    sentryEvent.SetExtra($"Exception[{i}][{key}]", sentryException.Data[key]);
+                    if (key.StartsWith("sentry", StringComparison.OrdinalIgnoreCase))
+                    {
+                        if (key.StartsWith(ExceptionDataTagKey, StringComparison.OrdinalIgnoreCase))
+                        {
+                            var data = sentryException.Data[key] as string ?? "<invalid>";
+                            //Set the key after the ExceptionDataTagKey string.
+                            sentryEvent.SetTag(key.Substring(ExceptionDataTagKey.Length), data);
+                        }
+                        else if (key.StartsWith(ExceptionDataContextKey, StringComparison.OrdinalIgnoreCase))
+                        {
+                            //Set the key after the ExceptionDataTagKey string.
+                            _ = sentryEvent.Contexts.AddOrUpdate(key.Substring(ExceptionDataContextKey.Length),
+                                sentryException.Data[key] ?? "<empty>",
```

Comment:
You can't use `<empty>` since `contexts` expects an object.

Since version 3.0.0 the SDK isn't serializing random objects anymore 
see: #822 

So we have a problem with this right now, but this must be an object, not a simple type.

Reply:
users can set null to Contexts like
`SentrySdk.WithScope(s => { s.Contexts.AddOrUpdate("", null, (_, value) => value); });`

mmm, should I just ignore if the object is null?

Reply:
Key wirhiut a value we can log and drop. 

Value as string will just render an error is whst I said. It's not supported

Reply:
In this case, I think the best scenario is to fall back to the original output as Exception[i][context] = data